app/lib/frontend/cronjobs.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:shelf/shelf.dart' show Request, Response;
+import 'package:logging/logging.dart' show Logger;
+import 'package:googleapis_auth/auth_io.dart' as auth;
+import 'package:googleapis/datastore/v1.dart' as datastore;
+import 'package:ulid/ulid.dart' show Ulid;
+
+final _log = new Logger('pub.cronjobs');
+
+class CronJobs {
+  final String backupBucketName;
+
+  CronJobs({
+    this.backupBucketName,
+  });
+
+  Future<Response> handler(Request request) async {
+    final job = _resolve(request.url.path);
+    if (job == null) {
+      return Response.notFound('no-such-cron-job');
+    }
+    _log.finest('### Cron: ${request.url.path}');
+    final ok = await job();
+    if (!ok) {
+      _log.shout('### Cron failed: ${request.url.path}');
+      return Response.internalServerError(
+        body: 'cron failed ${request.url.path}',
+      );
+    }
+    _log.finest('### Cron successful: ${request.url.path}');
+    return Response.ok('cron successful: ${request.url.path}');
+  }
+
+  /// Map from path to cronjob.
+  Future<bool> Function() _resolve(String path) {
+    return <String, Future<bool> Function()>{
+      '/cron/datastore-backup': datastoreBackup,
+    }[path];
+  }
+
+  /// Backup datastore
+  Future<bool> datastoreBackup() async {
+    if (backupBucketName == null) {
+      _log.shout('datastore backup not configured!!!');
+      return false;
+    }
+
+    // Get credentials via. metadata service (this only works on GCP/AppEngine)
+    final client = await auth.clientViaMetadataServer();
+    try {
+      // Create a unique job id, in case multiple jobs are trigger concurrently
+      final id = Ulid().toString();
```

Comment:
I like the use of ulid :), but we can get away using the uuid which we use elsewhere: as the file is prefixed by the date, and there will be a timestamp on the backup file too.

Reply:
I don't trust the uuid package it has phrases like:
>  * Cryptographically strong random number generation on all platforms
>    * Defaults to non-crypto generator, see UuidUtil for cryptoRNG

This might be different in 2.0.0, but we haven't upgraded to that yet, and the `README` for the uuid package still looks sketchy... :)